src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -158,23 +164,26 @@ void merge(ProcessConfig processConfig, ZonedDateTime dateTime, String date, Str
 
                 if (processConfig.isRunBalancesAdjustment()) {
                     // balances adjustment on the merge network
-                    balancesAdjustmentService.doBalance(networkUuids);
-
-                    LOGGER.info("Merge {} of process {}: balance adjustment complete", date, processConfig.getProcess());
+                    return balancesAdjustmentService.doBalance(networkUuids).flatMap(res -> {
+                        LOGGER.info("Merge {} of process {}: balance adjustment complete", date, processConfig.getProcess());
```

Comment:
.then instead of flatMap ?

Reply:
i don't see how to use then in this case, then(Mono<V> other)  then should return directly another mono but here we should execute two  instructions before returning that mono!

Reply:
You're right, can't use then(); Instead, use doOnSuccess ?